doc/source/whatsnew/v0.20.0.txt

```diff
@@ -193,6 +193,7 @@ Map on Index types now return other Index types
 Other API Changes
 ^^^^^^^^^^^^^^^^^
 
+- ``pd.read_csv()`` will now issue a ``UserWarning`` whenever there are conflicting values provided by the ``dialect`` parameter and the user (:issue:`14898`)
```

Comment:
is this the kind of warning we typically have for other parsing warnings?

maybe we should have a ParserWarning (subclass of UserWarning)?